python/strings_udf/cpp/include/cudf/strings/udf/udf_apis.hpp

```diff
@@ -26,14 +27,36 @@ namespace cudf {
 namespace strings {
 namespace udf {
 
+class udf_string;
+
 /**
  * @brief Return a cudf::string_view array for the given strings column
  *
+ * No string data is copied so the input column controls the lifetime of the
+ * underlying strings.
+ *
+ * New device memory is allocated and returned to hold just the string_view instances.
+ *
  * @param input Strings column to convert to a string_view array.
- * @throw cudf::logic_error if input is not a strings column.
+ * @return Array of string_view objects in device memory
  */
 std::unique_ptr<rmm::device_buffer> to_string_view_array(cudf::column_view const input);
 
+/**
+ * @brief Return a STRINGS column given an array of udf_string objects
+ *
+ * This will make a copy of the strings in d_string in order to build
+ * the output column.
+ * The individual udf_strings are also cleared freeing each of their internal
```

Comment:
What's the rationale for clearing as part of this function? The clearing seems orthogonal (and it makes the code less flexible) given that the column is copying the data out anyway.

Reply:
I agree. This was a convenience for Brandon's code. I could add a separate API to free the strings.